source/cellular_common.c

```diff
@@ -542,47 +540,35 @@ CellularError_t _Cellular_CreateSocketData( CellularContext_t * pContext,
         if( pContext->pSocketData[ socketId ] == NULL )
         {
             #if ( CELLULAR_CONFIG_STATIC_SOCKET_CONTEXT_ALLOCATION == 1 )
-                {
-                    pSocketData = &cellularStaticSocketDataTable[ socketId ];
-                }
-            #else
-                {
-                    pSocketData = ( CellularSocketContext_t * ) Platform_Malloc( sizeof( CellularSocketContext_t ) );
-                }
-            #endif
-
-            if( pSocketData != NULL )
-            {
+                pSocketData = &cellularStaticSocketDataTable[ socketId ];
+                cellularStatus = CELLULAR_SUCCESS;
                 createSocketSetSocketData( contextId, socketId, socketDomain,
                                            socketType, socketProtocol, pSocketData );
                 pContext->pSocketData[ socketId ] = pSocketData;
                 *pSocketHandle = ( CellularSocketHandle_t ) pSocketData;
-            }
-            else
-            {
-                cellularStatus = CELLULAR_NO_MEMORY;
-            }
+            #else
+                pSocketData = ( CellularSocketContext_t * ) Platform_Malloc( sizeof( CellularSocketContext_t ) );
+
+                if( pSocketData == NULL )
+                {
+                    LogError( ( "_Cellular_CreateSocketData, Out of memory." ) );
+                }
+                else
+                {
+                    cellularStatus = CELLULAR_SUCCESS;
+                    createSocketSetSocketData( contextId, socketId, socketDomain,
+                                               socketType, socketProtocol, pSocketData );
+                    pContext->pSocketData[ socketId ] = pSocketData;
+                    *pSocketHandle = ( CellularSocketHandle_t ) pSocketData;
+                }
+            #endif /* if ( CELLULAR_CONFIG_STATIC_SOCKET_CONTEXT_ALLOCATION == 1 ) */
 
             break;
         }
     }
 
     taskEXIT_CRITICAL();
 
-    if( cellularStatus == CELLULAR_NO_MEMORY )
-    {
-        LogError( ( "_Cellular_CreateSocket, Out of memory" ) );
-    }
-    else if( socketId >= CELLULAR_NUM_SOCKET_MAX )
-    {
-        LogError( ( "_Cellular_CreateSocket, No free socket slots are available" ) );
```

Comment:
I suggest to keep this error message if no free socket ID found for debugging.

Reply:
Update in new commit.